stone/frontend/ir_generator.py

```diff
@@ -319,6 +328,73 @@ def _create_type(self, env, item):
         env[item.name] = api_type
         return api_type
 
+    def _merge_patches(self):
+        """Injects object patches into their original object definitions."""
+        for patched_item, patched_namespace in self._patch_data_by_canonical_name.values():
+            patched_item_base_name = self._get_base_name(patched_item.name, patched_namespace.name)
+            if patched_item_base_name not in self._item_by_canonical_name:
+                raise InvalidSpec('Patch {} must correspond to a pre-existing data_type.'.format(
+                    quote(patched_item.name)), patched_item.lineno, patched_item.path)
+
+            existing_item = self._item_by_canonical_name[patched_item_base_name]
+
+            self._check_patch_type_mismatch(patched_item, existing_item)
+
+            if (isinstance(patched_item, AstStructPatch)
```

Comment:
Can compress it into: `isinstance(patched_item, (AstStructPatch, AstUnionPatch))`

Reply:
Done.